watertap/core/util/model_diagnostics/ipopt_initialization.py

```diff
@@ -0,0 +1,135 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import pyomo.environ as pyo
+from idaes.core.util.scaling import get_scaling_factor, __none_left_mult
+from idaes.core.solvers import get_solver
+
+
+def assert_no_initialization_perturbation(blk, optarg=None, solver=None):
+    """
+    Assert that IPOPT will *not* move the initialization
```

Comment:
Can you explain example use cases for the novel developer/user on when this tool would come in handy? When do I use this? How does it help me?